frontend/tests/controllers/CourseRegistrationTest.php

```diff
@@ -0,0 +1,155 @@
+
+    public static function setUpBeforeClass() {
+        parent::setUpBeforeClass();
+
+        $curatorGroup = \OmegaUp\DAO\Groups::findByAlias(
+            \OmegaUp\Authorization::COURSE_CURATOR_GROUP_ALIAS
+        );
+
+        [
+            'identity' => self::$curator,
+        ] = \OmegaUp\Test\Factories\User::createUser();
+        \OmegaUp\DAO\GroupsIdentities::create(
+            new \OmegaUp\DAO\VO\GroupsIdentities([
+                'group_id' => $curatorGroup->group_id,
+                'identity_id' => self::$curator->identity_id,
+            ])
+        );
+    }
+
+    private static function createCourseWithRegistrationMode() {
+        $adminLogin = self::login(self::$curator);
+        $school = SchoolsFactory::createSchool()['school'];
+        $alias = \OmegaUp\Test\Utils::createRandomString();
+
+        $response = \OmegaUp\Controllers\Course::apiCreate(
+            new \OmegaUp\Request([
+                'auth_token' => $adminLogin->auth_token,
+                'name' => \OmegaUp\Test\Utils::createRandomString(),
+                'alias' => $alias,
+                'description' => \OmegaUp\Test\Utils::createRandomString(),
+                'start_time' => (\OmegaUp\Time::get() + 60),
+                'finish_time' => (\OmegaUp\Time::get() + 120),
+                'school_id' => $school->school_id,
+            ])
+        );
+
+        $course = \OmegaUp\DAO\Courses::getByAlias($alias);
+
+        // Update to registration the admission mode
+        \OmegaUp\Controllers\Course::apiUpdate(new \OmegaUp\Request([
+            'auth_token' => $adminLogin->auth_token,
+            'course_alias' => $alias,
+            'name' => $course->name,
+            'description' => $course->description,
+            'alias' => $course->alias,
+            'admission_mode' => \OmegaUp\Controllers\Course::ADMISSION_MODE_REGISTRATION,
+        ]));
+
+        return [
+            'course' => $course,
+            'adminLogin' => $adminLogin,
+        ];
+    }
+
+    /**
+     * @param list<\OmegaUp\DAO\VO\Courses>  $coursesList
+     */
+    private function assertCourseIsPresentInArray($courses, string $alias) {
+        foreach ($courses as $course) {
+            if ($course['alias'] === $alias) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public function testCourseIsPresentInStudentList() {
+        $course = self::createCourseWithRegistrationMode()['course'];
+        ['identity' => $student] = \OmegaUp\Test\Factories\User::createUser();
+        $studentLogin = self::login($student);
+
+        // Course should appear in public course list for students
+        $coursesList = \OmegaUp\Controllers\Course::apiListCourses(
+            new \OmegaUp\Request([
+                'auth_token' => $studentLogin->auth_token,
+            ])
+        );
+
+        $this->assertCourseIsPresentInArray(
+            $coursesList['student'],
+            $course->alias
+        );
```

Comment:
```suggestion
        $this->assertArrayContainsWithPredicate(
            $coursesList['student'],
            /** ... */
            function ($studentCourse) use ($course): bool {
                return $studentCourse['alias'] == $course->alias;
            }
        );
```

así no se necesita declarar `assertCourseIsPresentInArray`!

Reply:
Listo